types/prettier/index.d.ts

```diff
@@ -344,6 +344,7 @@ export interface SupportOptionRange {
 export type SupportOptionType = 'int' | 'boolean' | 'choice' | 'path';
 
 export interface BaseSupportOption<Type extends SupportOptionType> {
+    name: string;
```

Comment:
Are you sure you modifying the correct thing?
You point to a URL: https://prettier.io/docs/en/api.html#prettiergetsupportinfo
But this only explains/describes declaration for `SupportLanguage` in line 322
And there is the `name` property in line [323](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50238/files#diff-cc4da3b15bc8ed3b59bf35e45562c3f182230a9330f787ac982266821c8efa85R323) :thinking:

Reply:
I think the URL's info is incorrect cuz `getsupportinfo` return object `{languages: [{...}], options: [{...}]}`

Please check my reproduct code https://codesandbox.io/s/prettier-wveu3?file=/src/index.ts
Type `SupportOption ` relate to  `BaseSupportOption` .

```ts
export type SupportOption =
    | IntSupportOption
    | IntArraySupportOption
    | BooleanSupportOption
    | BooleanArraySupportOption
    | ChoiceSupportOption
    | PathSupportOption
    | PathArraySupportOption;

export interface IntSupportOption extends BaseSupportOption<'int'> { // extends BaseSupportOption
    default: number;
    array?: false;
    range?: SupportOptionRange;
}
```

But not exist `name`.
```ts
export interface BaseSupportOption<Type extends SupportOptionType> {
    since: string;
    category: string;
    /**
     * The type of the option.
     *
     * When passing a type other than the ones listed below, the option is
     * treated as taking any string as argument, and `--option <${type}>` will
     * be displayed in --help.
     */
    type: Type;
    /**
     * Indicate that the option is deprecated.
     *
     * Use a string to add an extra message to --help for the option,
     * for example to suggest a replacement option.
     */
    deprecated?: true | string;
    /**
     * Description to be displayed in --help. If omitted, the option won't be
     * shown at all in --help.
     */
    description?: string;
}
```

This is strange for me.

Reply:
Mh okay :thinking:  
But adding the property in the way you did, would e.g. break my plugin
https://github.com/prettier/plugin-pug/blob/3deaa53a5b5fdcd12ffd102f42b00e7789c1ebf5/src/options/attribute-separator.ts#L4-L27

`name` is definitely not a required property...

So maybe we could try
```suggestion
    readonly name?: string;
```

I think it's readonly cause you only can/should read the value. But providing a new option from a plugin the name is the key of [`SupportOptions`](https://github.com/tyankatsu0105/DefinitelyTyped/blob/f80a5430a990ae19b282179b7671326dd37e352b/types/prettier/index.d.ts#L424) and this is used here https://github.com/tyankatsu0105/DefinitelyTyped/blob/f80a5430a990ae19b282179b7671326dd37e352b/types/prettier/index.d.ts#L177

Reply:
OK. Thank for the advice.

Reply:
done https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50238/commits/26709c575efe503259ed94ae95c73b980986fdd3